tests/framework/e2e/cluster.go

```diff
@@ -491,6 +491,10 @@ func (epc *EtcdProcessCluster) EndpointsV3() []string {
 	return epc.Endpoints(func(ep EtcdProcess) []string { return ep.EndpointsV3() })
 }
 
+func (epc *EtcdProcessCluster) EndpointsGRPC() []string {
```

Comment:
I'm fine backporting https://github.com/etcd-io/etcd/commit/65add8cec44b32682e59d2596a592038bac90dcd in full. @ahrtr, thoughts?

Reply:
It isn't a big deal to me. I think it's accepted to do it in a separate PR.

Reply:
As I was backporting that commit, I discovered that almost everything from that commit had already been backported by commit 795dcedd1d. However, the only functions not backported by that commit were defining `EtcdClusterProcess`'s `EndpointsGRPC()` and `EndpointsHTTP()`.

So, this is the diff from backporting that commit.

```diff
diff --git a/tests/framework/e2e/cluster.go b/tests/framework/e2e/cluster.go
index 97d584571..d18d0c88c 100644
--- a/tests/framework/e2e/cluster.go
+++ b/tests/framework/e2e/cluster.go
@@ -491,6 +491,14 @@ func (epc *EtcdProcessCluster) EndpointsV3() []string {
        return epc.Endpoints(func(ep EtcdProcess) []string { return ep.EndpointsV3() })
 }
 
+func (epc *EtcdProcessCluster) EndpointsGRPC() []string {
+       return epc.Endpoints(func(ep EtcdProcess) []string { return ep.EndpointsGRPC() })
+}
+
+func (epc *EtcdProcessCluster) EndpointsHTTP() []string {
+       return epc.Endpoints(func(ep EtcdProcess) []string { return ep.EndpointsHTTP() })
+}
+
 func (epc *EtcdProcessCluster) Endpoints(f func(ep EtcdProcess) []string) (ret []string) {
        for _, p := range epc.Procs {
                ret = append(ret, f(p)...)
```

I think it would be better to split the commit into two rather than raise a PR for that change. Or, @jmhbnz, would you prefer a new PR?

Reply:
Happy to do it here in this pr, I think it should just be separate commit to the other backport 🙂

Reply:
:facepalm: sorry, James, I didn't thoroughly read your second paragraph.
Just a heads up that it seems 3.4 also doesn't have defined these two functions.